signer/Cargo.toml

```diff
@@ -26,34 +27,36 @@ ecdsa = ["secp256k1"]
 
 # Make the keypair algorithms here compatible with Subxt's Signer trait,
 # so that they can be used to sign transactions for compatible chains.
-subxt = ["dep:subxt"]
+subxt = ["dep:subxt-core"]
 
 # The getrandom package is used via schnorrkel. We need to enable the JS
 # feature on it if compiling for the web.
-web = ["getrandom/js", "subxt?/web"]
-native = ["subxt?/native"]
+web = ["getrandom/js"]
 
 [dependencies]
-subxt = { workspace = true, optional = true, default-features = false }
-regex = { workspace = true }
+subxt-core = { workspace = true, optional = true, default-features = false }
+regex = { workspace = true, default-features = false }
 hex = { workspace = true }
 codec = { package = "parity-scale-codec", workspace = true, features = ["derive"] }
-sp-core-hashing = { workspace = true }
-thiserror = { workspace = true }
-pbkdf2 = { workspace = true }
-sha2 = { workspace = true }
-hmac = { workspace = true }
+sp-core-hashing = { workspace = true, default-features = false }
+derive_more = { workspace = true }
+pbkdf2 = { workspace = true, default-features = false }
+sha2 = { workspace = true, default-features = false }
+hmac = { workspace = true, default-features = false }
 zeroize = { workspace = true }
-bip39 = { workspace = true }
-schnorrkel = { workspace = true, optional = true }
-secp256k1 = { workspace = true, features = ["recovery", "global-context"], optional = true }
+bip39 = { workspace = true, default-features = false, features = ["unicode-normalization"] }
+schnorrkel = { workspace = true, optional = true, default-features = false }
+secp256k1 = { workspace = true, optional = true, default-features = false, features = ["alloc", "recovery"] } # features = ["recovery", "global-context"], 
 secrecy = { workspace = true }
 
+# We only pull this in because `std::sync::OnceLock` is not available in no-std contexts.
+once_cell = { workspace = true, default-features = false, features = ["alloc", "critical-section"] }
```

Comment:
`critical-section` uses this crate to provide the only-one-thread-at-a-time behaviour:

https://crates.io/crates/critical-section

This crate requires that you enable different features to support different no-std targets (perhaps this explains why you hit a compile error in no-std).

I'm not sure whether this is a good idea to use or not. 

If we do use it, we should def enable `critical-section/std` when `std` feature is enabled though, so that it uses a standard mutex instead of anything strange in std cases. We'd also need to document to users to go read the `critical-section` docs (like once_cell does) because they would need to depend on `critical-section` themselves and add the relevant features for the targets they are trying to support.

But, all of this is just to enable the provided dev accounts at the end of the day, so perhaps the best route to take is revert to using `std::sync::OnceLock`, remove `once_cell` dep and just don't globally instantiate the dev accounts when in no-std mode. So we could alter the `once_static` macro thing to just use the OnceLock global when in std and recreate the account (run the `$expr`) every time when in no-std? (and then document on the macro that in no-std we run the expr every time)